### PR TITLE
Added unit tests for 3 functions in NoteService

### PR DESCRIPTION
**Pull request by TheItsyBitsySpider**
## Purpose / Description
Added 4 unit tests to increase code coverage of NoteService.

According to the coverage report, it should completely cover updateJsonNoteFromMultimediaNote and saveMedia, and cover most of importMediaToDirectory

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    public void updateJsonNoteRuntimeErrorTest() {
+        //model with ID 42
+        Model testModel = new Model("{\"flds\": [{\"name\": \"foo bar\", \"ord\": \"1\"}], \"id\": \"42\"}");
+        IMultimediaEditableNote multiMediaNoteWithID42 = NoteService.createEmptyNote(testModel);
+
+        //model with ID 45
+        testModel = new Model("{\"flds\": [{\"name\": \"foo bar\", \"ord\": \"1\"}], \"id\": \"45\"}");
+        Note noteWithID45 = new Note(testCol, testModel);
+
+        assertThrows(RuntimeException.class, () -> NoteService.updateJsonNoteFromMultimediaNote(multiMediaNoteWithID42, noteWithID45));
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.java:71]: `assertThrows` should return a `Throwable`, could you add an `assert` on the message

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    //similar test like above, but with an imagefield instead of an audioclipfield
+    @Test
+    public void importImageToDirectoryTest() throws IOException {
+
+        File mFile = folder.newFile("testimage.png");
+
+        //writes a line in the file so the file's length isn't 0
+        FileWriter fileWriter = new FileWriter(mFile);
+        fileWriter.write("line1");
+        fileWriter.close();
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.java:110]: Could you use a `try with resources` here

File: AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        ImageField imgField = new ImageField();
+        imgField.setImagePath(mFile.getAbsolutePath());
+        testImage.setField(0, imgField);
+
+        NoteService.saveMedia(testCol, testImage);
+
+        File outFile = new File(testCol.getMedia().dir(), mFile.getName());
+
+        //now path should be equal to the new file made in NoteService.saveMedia
+        assertEquals(outFile.getAbsolutePath(), imgField.getImagePath());
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/services/NoteServiceTest.java:124]: optional: You can move the comment to a string in the assert

**Review by david-allison-1 — COMMENTED** [review]
Looks great! Thanks

Only nitpicks

**Review by david-allison-1 — APPROVED** [review]
Cheers! LGTM

**Comment by david-allison-1** [comment]
note to maintainers: squash merge

**Comment by david-allison-1** [comment]
@TheItsyBitsySpider Just to note: we aim to have a linear history on `master` (no merge commits). 

For future branches after making review changes, Could you `rebase + push --force-with-lease` if yo want to move to the latest `master`) , instead of merging. It allows us to `rebase` rather than `squash`.

Note: not necessary for this one, as the change should be a single commit anyway, so it's fine that e squash this.

PS: Thanks again! Testing improvements are fantastic. I put a lot of time into unit testing, and it's great to see others doing the same

**Comment by TheItsyBitsySpider** [comment]
Sure! I'll avoid doing merge commits for future branches, and I'll use the `rebase + push --force-with-lease` method instead.

Also, I'm glad that you like my unit tests! Planning on doing more over the summer, and I'm happy that it's helping out.

**Review by mikehardy — APPROVED** [review]
Nice - thank you!

**Pull request merged**